articles/npm-vs-npx-what-is-the-difference/index.md

```diff
@@ -0,0 +1,269 @@
+
+```bash
+./node_modules/.bin/your-package-name
+```
+
+```bash
+./node_modules/.bin/eslint --init
+```
+
+And provide the argument to execute
+
+```bash
+./node_modules/.bin/eslint yourfile.js
+```
+
+NPX comes in to solve a couple of keystrokes to make it easier to execute Node.js modules. It makes it easier to run and interact with executables hosted in the NPM registry. It makes it easier to executed NPM local binaries. It comes with many added features.
+
+### Using NPX
+
+With NPX, you can run and executes packages without having to install them locally or globally.
+
+When running NPM executables with NPX, if a package is installed, NPX will search for the package binaries (either local or global) and then run the package.
+
+If the package is not installed, NPX will download and cache the package directly from the NPM registry without locally saving the package binaries. It will execute the package binaries without installing the packages in your system, as you would typically do with NPM.
+
+Let's check some key use cases of NPX.
+
+Run the following command to make sure you have NPX installed using `which npx`.
+
+If not installed, use `npm install -g NPM@latest` to update NPM to the higher version above 5.2 or later, and you will have the NPX CLI tool available. You can also install NPX as a stand-alone package. Run `npm i npx` to install the NPX runner.
+
+#### NPX standard command
+
+With NPX, you just need this NPX command to execute the package.
+
+```bash
+npx your-package
+```
+
+Type `npx --help` to get a more detailed npx commands syntax.
+
+```bash
+  npx [options] <command>[@version] [command-arg]...
+
+  npx [options] [-p|--package <package>]... <command> [command-arg]...
+
+  npx [options] -c '<command-string>'
+```
+
+#### Executing installed packages
+
+When using NPM, you had to remember and type the package path to the command, as explained in the Eslint case.
+
+```bash
+./node_modules/.bin/eslint yourfile.js
+```
+
+This is tiresome to always add the node modules path, referencing them all over your files and commands.
+
+With NPX, you can have your local packages installed and use NPX commands (instead of adding the whole path) to execute the package binaries.
+
+```bash
+npx eslint yourfile.js
+```
+
+NPX will execute packages without having to install them. It will create a temporary cache and then remove it when the execution is complete. This saves on disk size and allows you to run a package only when it's needed. It also gives you the advantage of testing packages without having to install them.
+
+NPX lets you execute local 'node modules' as if they were global. If the package was not previously installed, NPX would not install the package in your system; instead, it will create a temporary cache that will hold the package binaries. Once the execution is over, NPX will remove the installed cache binaries from the system. This way, your globals stays clean.
+
```

Comment:
Some points are very repetitive, as the temporary cache without installing. Make sure you don't repeat points.